client-common/src/main/java/org/apache/cassandra/sidecar/common/ApiEndpointsV1.java

```diff
@@ -118,6 +118,11 @@ public final class ApiEndpointsV1
 
     public static final String CONNECTED_CLIENT_STATS_ROUTE = API_V1 + CASSANDRA + "/stats/connected-clients";
 
+    // Endpoint to retrieve sstable's preemptiveOpenInterval value.
+    // Value returned is in MB, may return negative value when disabled
+    private static final String SSTABLE = "/sstable";
+    public static final String SSTABLE_PREEMPTIVE_OPEN_INTERVAL_ROUTE = API_V1 + CASSANDRA + SSTABLE +
```

Comment:
I know we are reusing the `getSsTablePreemtiveOpenIntervalInMB` JMX operation, but, from a pure API standpoint, we are omitting the `MB` part in the endpoint, but returning it as part of the response (`SSTablePreemptiveOpenIntervalInMB`). My suggestion for this would be to either:
- We add the `MB` part to the endpoint (`/preemptive-open-interval-in-mb`)
Or
- We just support an optional `unit` filtering parameter in the url, that can be extended in the future with other unit types if needed. That way, the url would end in `/preemptive-open-interval?unit=mb` (remember that the unit is optional with MB as a default, so no need to pass it from the client).

That would make us have to modify the response to something like:
```
{
  "SSTablePreemptiveOpenInterval": 1234,
  "unit": "MB"
}
```


What do you think?

Reply:
- We want to make endpoint generic, so underlying changes to C* shouldn't impact the endpoint usage for the end user, for example C* 6.0 returning value in different unit, while older versions continue to return the value in MB, the same endpoint should work regardless of the running c* version. Hence not including MB in the endpoint name. Whereas JMX operation name need to match exactly the JMX endpoint we are invoking. The same with passing option param like unit=MB, better we avoid supporting new unit param in the endpoint whenever C* changes for different versions.

- In the response we need to somehow tell the end user that value returned is in in MB. I preferred the name SSTablePreemptiveOpenIntervalInMB as it matches the JMX operation currently the end users are calling. Your suggestion of separating the unit into another field sounds good, but that may complicate callers's code, as they need to parse unit as a string and convert value into those units accordingly. C* latest code changed this yaml param to for example "60MiB", let me see which one is simpler for the calling code "60MiB" or {60, "MiB"} and will get back to you

Reply:
I did explore this a little, I believe the one you suggested /preemptive-open-interval?unit=mb is more easy way to implement and also is in-line with current JMX call semantics. Other alternatives unnecessarily complicates conversions from human readable format to byte size at the client.

Reply:
Added unit=MiB param, unit names matches with names C* is using.